### PR TITLE
Fix LoggingMixin to work with subclasses

### LINKED ISSUE #201: Better logging utility
[`LoggingMixin`](https://github.com/ppb/pursuedpybear/blob/46ab2a3c145b0c41a3e7d1db481573d26249c465/ppb/utils.py#L4-L16) is just... woefully naive in the face of object heirarchies.

We need something that respects where the function is defined.

Problem is, getting a stack trace in python is slow. Like, `inspect.stack()` is a half millisecond per call. It doesn't include the class name anyway. (Just the function and file name.)

Adding values to a function's globals requires some type magic. (Not to mention the work involved in overriding a single global variable on a per-function basis while letting setting global values still work.) Adding a function closure is basically black magic.

https://www.youtube.com/watch?v=EiOglTERPEo might have some insights, but I don't have time to watch right now.

### PR DESCRIPTION
**Pull request by astronouth7303**
Fixes #201

### CODE REVIEW

**Comment by astronouth7303** [comment]
Unfortunately, this reduces the logger tree from `package.module.Class` to `package.module`, but I don't think that's a major problem.

File: ppb/utils.py [hunk 1]
```diff
@@ -1,17 +1,48 @@
...
-    _logger = None
 
     @property
     def logger(self):
-        if self._logger is None:
-            fqn = f"{self.__class__.__module__}.{self.__class__.__qualname__}"
-            self._logger = logging.getLogger(fqn)
-        return self._logger
+        # This is internal/CPython only/etc
+        # It's also astonishingly faster than alternatives.
```
[the remark this answers is not in the document]
Comment (astronouth7303) [reply ppb/utils.py]: It's an implementation detail and this information doesn't make sense in the doc string.

[the remark this answers is not in the document]
Comment (pathunstrom) [reply ppb/utils.py]: I'll accept that answer.

File: ppb/utils.py [hunk 2]
```diff
@@ -1,17 +1,48 @@
...
+    }
+
+
+def _get_module(file_name):
+    """
+    Find the module name for the given file name, or raise KeyError if it's
+    not a loaded module.
+    """
+    if file_name not in _module_file_index:
+        _build_index()
```
[the remark this answers is not in the document]
Comment (astronouth7303) [reply ppb/utils.py]: That would be my assumption. First use in a given file should be relatively slow and after that, much quicker.

Of course, given the complexity of the operation, I don't expect it to actually be slow.

**Pull request merged**